### ISSUE TITLE
[bug] lock create ignore build options (conan 1.x)

### ISSUE THREAD

**Issue by hekrdla** [conan-io/conan#15078]
### Environment details

Operating System+version: windows 10
Compiler+version: msvc v142
Conan version: 1.61.0
Python version: 3.10.10

### Steps to reproduce

Hi I found that when using `conan lock create` with some package in both host and build context with different options, package in build context in lock file have options from host context rather that option defined for build context. 
Probably same issue happen when using package in build context with non default option.

Example:
```
conan lock create ... --options:host protobuf:shared=False --options:build protobuf:shared=True
```

I debug into [extend_build_requires](https://github.com/conan-io/conan/blob/develop/conans/client/graph/graph_builder.py#L73), here are created `new_options` that will later applied on package to create lock graph.
```
new_options = node.conanfile.build_requires_options._reqs_options
```
Only assignment to `build_requires_options` is in [graph_binaries.py](https://github.com/conan-io/conan/blob/develop/conans/client/graph/graph_binaries.py#L338). Based on that, these are backup of host options. But build options stored in `profile_build` are never accessed in `extend_build_requires`.
```
        # There might be options that are not upstream, backup them, might be for build-requires
        conanfile.build_requires_options = conanfile.options.values
```

I try to apply build options from `profile_build` on `new_options` [here](https://github.com/conan-io/conan/blob/develop/conans/client/graph/graph_builder.py#L83) and it works as expected
```
        if profile_build:  # Apply the fix only for the 2 profile scenario
            for k, v in profile_build.options._reqs_options.items():
                if k in new_options:
                    new_options[k].update(v)
                else:
                    new_options[k] = v
```

I'm not sure if this is proper fix and how to work with Options in profile without accessing private `_reqs_options` member.

### Note

I'm using conanfile.py as meta-package instead of conanfile.txt and it will be nice if build options can be accessed in `config_options` like host options (`self.options[<package>].<option-name> = value`). Similar to `settings` vs `settings_build` attribute.

**Comment by memsharded** [comment]
Hi @hekrdla 

Thanks for your detailed report and your deep investigation.
As you did the effort, it would be good if you can open a PR with that change, and we will see what happens. My concern is that this kind of change is typically very risky, and it is very likely that something can break for other users, so these changes are unlikely to happen in Conan 1.X releases.

> I'm using conanfile.py as meta-package instead of conanfile.txt and it will be nice if build options can be accessed in config_options like host options (self.options[<package>].<option-name> = value). Similar to settings vs settings_build attribute.

This has already been implemented in Conan 2.0, it is possible to define options for "build", both in ``default_build_options`` and also in ``self.tool_requires(..., options=...)``. It is also done only in 2.0 because of the same reasons, doing this in Conan 1.X had become increasingly complicated and risky.

**Comment by memsharded** [comment]
Closing as solved in Conan 2, but please create new tickets for any further question you might have. Thanks for the feedback.
